barcalendar.py

```diff
@@ -22,9 +22,87 @@
 import re
 import time
 
+import atexit
+import logging
 import requests
 import yaml
 
+def setup_custom_logging():
+    """
+    Creating a custom logger to flush all Warning logs at the end of the script
+    to avoid breaking the console script being generated by the script
+    """
+    class DelayedLogHandler(logging.Handler):
+        def __init__(self):
+            super().__init__()
+            self.log_messages = []
+
+        def emit(self, record):
+            self.log_messages.append(record)
+
+    def _flush_log_messages(handler):
+        for message in handler.log_messages:
+            print(message)
```

Comment:
If you prefix the message with `// ` then the full output will still be valid JavaScript, and could be copied into the spreadsheet without worrying about the log messages.

Also, these are records, not messages.  IDK if you want to extract the message to have a more readable output.

Reply:
Updated as per the suggestions.